### PR TITLE
Code review for Mainnet

### PR DESCRIPTION
**Pull request by themue**
Researched if we need:

- [x] special targets to build for Mainnet (`statusgo-mainnet`, flags `mainnet-flags.sh`)
- [x] global variables `UseMainnet` and `UseMainnetFlag`

Code and build already prepared for Mainnet, only small documentation in testing package `t` needed.

Solves #699

### CODE REVIEW

File: t/utils/utils.go [hunk 1]
```diff
@@ -170,7 +171,7 @@ func GetHeadHashFromNetworkID(id int) string {
 	case params.StatusChainNetworkID:
 		return "0xe9d8920a99dc66a9557a87d51f9d14a34ec50aae04298e0f142187427d3c832e"
 	}
-
+	log.Debug("invalid network id", "networkid", id)
```
Comment (mandrigin) [inline t/utils/utils.go]: Maybe some specific log with error for mainnet. So if anyone misuses the API he sees a useful message, like "tests shouldn't be run on the main network".

File: t/utils/utils.go [hunk 2]
```diff
@@ -195,7 +197,7 @@ func GetNetworkID() int {
 	case fmt.Sprintf("%d", params.StatusChainNetworkID), "statuschain":
 		return params.StatusChainNetworkID
 	}
-
+	log.Debug("invalid network id, fallback to statuschain", "networkid", *networkSelected)
```
Comment (mandrigin) [inline t/utils/utils.go]: The same here, add a separate error message for the main net, so someone who misuses the code at knows that that's intentional.

**Review by mandrigin — APPROVED** [review]
LGTM!

2 optional comments.

**Comment by dshulyak** [comment]
i thought that we want to have a subset of tests that can be executed on mainnet, it is not the case?

**Comment by themue** [comment]
@dshulyak Yes, that's a different task, see https://github.com/orgs/status-im/projects/15 in *To Do*.

**Review by adambabik — COMMENTED** [review]
Not really sure if a log is enough. The test may silently pass and even though the error log is written to stdout, it won't be displayed. 

`panic()` maybe? I am also not a fan of this approach but have no better idea...

**Comment by themue** [comment]
As a full stop `panic()` is indeed better, yes. So will change it.

**Comment by divan** [comment]
@themue what's about global variables `UseMainnet` and `UseMainnetFlag` mentioned in PR description?
And I second the opinion on not using Debug/Logs for tests (or, at least, using them only for exceptional cases).

**Comment by themue** [comment]
@divan So not only `panic()` when Mainnet is passed as argument but for any illegal argument?

We currently set variables `UseMainnet` and `UseMainnetFlag` during build. See `config.go` line 18ff. But it looks like `UseMainnet` is used nowhere. So thanks for hint.

File: geth/params/config.go [hunk 3]
```diff
@@ -430,9 +430,15 @@ func (c *NodeConfig) Save() error {
 	return nil
 }
 
-// updateConfig traverses configuration and adjusts dependent fields
+// validateAndUpdateConfig traverses configuration and adjusts dependent fields
 // (we have a development/production and mobile/full node dependent configurations)
-func (c *NodeConfig) updateConfig() error {
+func (c *NodeConfig) validateAndUpdateConfig() error {
+	// Validate config
+	if !UseMainnet && c.NetworkID == MainNetworkID {
```
Comment (adambabik) [inline geth/params/config.go]: Maybe we can have a unit test for this condition?

**Review by adambabik — APPROVED** [review]
LGTM, one request for a unit test.

**Comment by adambabik** [comment]
@themue actually, we have a unit test that caught that:
```
--- FAIL: TestLoadNodeConfig (0.02s)
	config_test.go:371: /tmp/geth-config-tests307058677
	config_test.go:374: test: invalid input JSON (missing comma at the end of key:value pair)
	config_test.go:374: test: check static DataDir passing
	config_test.go:374: test: use default KeyStoreDir
	config_test.go:374: test: use non-default KeyStoreDir
	config_test.go:374: test: test Upstream config setting
	config_test.go:374: test: test parameter overriding
	config_test.go:374: test: test loading Testnet config
	config_test.go:374: test: test loading Mainnet config
	Error Trace:	config_test.go:184
			config_test.go:377
	Error:      	Received unexpected error:

```

**Comment by themue** [comment]
As discussed with @adambabik will remove the flag and check for mainnet. Hasn't been used before and creates too much effort with no need.

**Comment by themue** [comment]
@adambabik Without the flags it's working now.

File: Makefile [hunk 4]
```diff
@@ -47,7 +47,7 @@ HELP_FUN = \
 		   }
 
 statusgo: ##@build Build status-go as statusd server
-	go build -i -o $(GOBIN)/statusd -v -tags '$(BUILD_TAGS)' $(shell _assets/build/testnet-flags.sh) ./cmd/statusd
```
Comment (adambabik) [inline Makefile:50]: We may leave this one but remove `-X github.com/status-im/status-go/geth/params.UseMainnetFlag=false` part.

File: Makefile [hunk 5]
```diff
@@ -143,15 +126,15 @@ test-unit-coverage: ##@tests Run unit and integration tests with coverage
...
 	# order: reliability then alphabetical
 	# TODO(tiabc): make a single command out of them adding `-p 1` flag.
-	go test -timeout 5m ./t/e2e/accounts/... -network=$(networkid)
-	go test -timeout 5m ./t/e2e/api/... -network=$(networkid)
-	go test -timeout 5m ./t/e2e/node/... -network=$(networkid)
-	go test -timeout 50m ./t/e2e/jail/... -network=$(networkid)
-	go test -timeout 20m ./t/e2e/rpc/... -network=$(networkid)
-	go test -timeout 20m ./t/e2e/whisper/... -network=$(networkid)
-	go test -timeout 10m ./t/e2e/transactions/... -network=$(networkid)
+	go test -timeout 5m ./t/e2e/accounts/... -network=$(or $(networkid),StatusChain)
```
Comment (adambabik) [inline Makefile]: We can set it like this to have a default value:
```
GO ?= StatusChain
```

File: _assets/build/testnet-flags.sh [hunk 6]
```diff
@@ -1,13 +0,0 @@
...
-set -e
-
-if [ ! -f "_assets/build/testnet-flags.sh" ]; then
-    echo "$0 must be run from the root of the repository."
-    exit 2
-fi
-
-# set gitCommit when running from a Git checkout.
-if [ -f ".git/HEAD" ]; then
-    echo "-ldflags '-X github.com/status-im/status-go/geth/params.UseMainnetFlag=false -X main.buildStamp=`date -u '+%Y-%m-%d.%H:%M:%S'` -X main.gitCommit=$(git rev-parse HEAD)'";
```
Comment (adambabik) [inline _assets/build/testnet-flags.sh:12]: I would keep this file for now but remove `-X github.com/status-im/status-go/geth/params.UseMainnetFlag=false`.

**Review by adambabik — CHANGES_REQUESTED** [review]

File: Makefile (hunk 5 above)
Comment (themue) [reply Makefile]: Oh, nice, more elegant.

File: Makefile (hunk 4 above)
Comment (themue) [reply Makefile:50]: As we don't separate between mainnet and testnet compiling anymore I would prefer to set the optional flags as Makefile variable.

File: _assets/build/testnet-flags.sh (hunk 6 above)
Comment (themue) [reply _assets/build/testnet-flags.sh:12]: As mentioned above this part could be direct part of the Makefile.

File: Makefile (hunk 4 above)
Comment (adambabik) [reply Makefile:50]: Ok, that makes sense.

**Review by adambabik — APPROVED** [review]
LGTM

**Pull request merged**